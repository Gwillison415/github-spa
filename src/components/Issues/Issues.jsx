import React, { Component } from 'react';
import './Issues.css'
import { Card, Feed, List } from 'semantic-ui-react';
import Issue from '../Issue/Issue.jsx';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {retrieveIssues} from '../../actions/issueActions';

class Issues extends Component{

  componentDidMount(){
    this.props.retrieveIssues();
  }

  issueComponents = (issueIds) => {
    return issueIds.map(id => (
        <Issue key={id} issueId={id} />
    ))
  }

  render() {
    return (
      <Card>
        <Card.Content>
          <Card.Header className="ui center aligned">
            Git Issues
          </Card.Header>
        </Card.Content>
        <Card.Content>
          <Feed>
            <List divided relaxed>
              {this.issueComponents(this.props.issuesIds)}
            </List>
          </Feed>
        </Card.Content>
      </Card>
    )
  }
};

const mapStateToProps = state => {
    return {
      issuesIds: state.issues.ids,
      issuesById : state.issues.issuesById
    }
}

const mapDispatchToProps = dispatch => bindActionCreators ({
  retrieveIssues
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Issues);
