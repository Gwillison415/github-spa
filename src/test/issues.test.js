/* eslint-disable no-unused-vars, react/jsx-filename-extension */

import React from 'react';
// import renderer from 'react-test-renderer';
// import { withRouter, Route } from 'react-router-dom';

import toJson, { shallowToJson } from 'enzyme-to-json';
import { shallow, render, mount } from 'enzyme';
import {Issues, mapStateToProps, mapDispatchToProps }from '../components/Issues/Issues';
import retrieveIssues from '../actions/issueActions';

const state = {
  issues : {
    ids: [246869449],
    issuesById : {
      246869449 : {
        assignedTo: [],
        labels : [],
        repoIssueNumber: 36,
        title : "disregard this PR",

      }
    },
  },
  loadingIssues: false
}

describe('Issues component', () => {
  it('should render a component with props as specified ', () => {
    const component = shallow(
      <Issues issuesIds={state.issues.ids} issuesLoading={false}/>,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('map\'s given State To Props ', () => {

    const expected = {
      issuesIds: [246869449],

      loadingIssues: false
    }
    expect(mapStateToProps(state)).toEqual(expected);
  })

  it('maps component dispatches to props', () =>{
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('retrieveIssues');
  });
});
