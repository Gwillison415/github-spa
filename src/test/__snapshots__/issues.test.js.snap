// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issues component should render a snapshot component as specified  1`] = `
<Card>
  <CardContent>
    <CardHeader>
      Git Issues
    </CardHeader>
  </CardContent>
  <CardContent>
    <Feed>
      <List
        divided={true}
        relaxed={true}
      >
        <ListItem>
          <ListContent>
            <ListHeader>
              #1 Git Issue One
            </ListHeader>
            <LabelGroup
              tag={true}
            >
              <div
                className="us tiny label"
              >
                <Label>
                  question
                </Label>
              </div>
            </LabelGroup>
            <ListDescription>
              Assigned To: Jenny
            </ListDescription>
          </ListContent>
        </ListItem>
        <ListItem>
          <ListContent>
            <ListHeader>
              #2 Git Issue One
            </ListHeader>
            <LabelGroup
              tag={true}
            >
              <Label>
                bug
              </Label>
            </LabelGroup>
            <ListDescription>
              Assigned To: Jenny
            </ListDescription>
          </ListContent>
        </ListItem>
        <ListItem>
          <ListContent>
            <ListHeader>
              #3 Git Issue One
            </ListHeader>
            <LabelGroup
              tag={true}
            >
              <Label>
                help
              </Label>
            </LabelGroup>
            <ListDescription>
              Assigned To: Jenny
            </ListDescription>
          </ListContent>
        </ListItem>
      </List>
    </Feed>
  </CardContent>
</Card>
`;
